### PR TITLE
[Storage] Updated close handles

### PR DESCRIPTION
**Pull request by annatisch**

### CODE REVIEW

**Comment by adxsdk6** [comment]
Can one of the admins verify this patch?

File: sdk/storage/azure-storage-file/azure/storage/file/aio/directory_client_async.py [hunk 1]
```diff
@@ -262,50 +261,61 @@ def list_handles(self, recursive=False, **kwargs):
...
-            timeout=timeout,
-            sharesnapshot=self.snapshot,
-            recursive=recursive,
-            cls=return_response_headers,
-            **kwargs)
-        try:
-            start_close = await command()
-        except StorageErrorException as error:
-            process_storage_error(error)
+            handle_id = handle
```
Comment (rakshith91) [inline sdk/storage/azure-storage-file/azure/storage/file/aio/directory_client_async.py]: Should we make sure handle_id is not '*' here...and may be force users to use close_all_handles if they want to delete everything

File: sdk/storage/azure-storage-file/azure/storage/file/aio/directory_client_async.py [hunk 2]
```diff
@@ -262,50 +261,61 @@ def list_handles(self, recursive=False, **kwargs):
...
+                handle_id,
+                timeout=timeout,
+                marker=continuation_token,
+                sharesnapshot=self.snapshot,
+                cls=return_response_headers,
+                **kwargs
+            )
+            continuation_token = response.get('marker')
+            try_close = bool(continuation_token)
+            total_handles += response.get('number_of_handles_closed')
```
Comment (rakshith91) [inline sdk/storage/azure-storage-file/azure/storage/file/aio/directory_client_async.py]: should we move this given this method is supposed to be used for closing only one handle?

File: sdk/storage/azure-storage-file/azure/storage/file/aio/directory_client_async.py (hunk 1 above)
Comment (annatisch) [reply sdk/storage/azure-storage-file/azure/storage/file/aio/directory_client_async.py]: Yup - will add.

File: sdk/storage/azure-storage-file/azure/storage/file/aio/directory_client_async.py (hunk 2 above)
Comment (annatisch) [reply sdk/storage/azure-storage-file/azure/storage/file/aio/directory_client_async.py]: Yup - will fix

File: sdk/storage/azure-storage-file/azure/storage/file/aio/file_client_async.py [hunk 3]
```diff
@@ -864,40 +863,71 @@ def list_handles(self, **kwargs):
...
+                    **kwargs
+                )
+            except StorageErrorException as error:
+                process_storage_error(error)
+            continuation_token = response.get('marker')
+            try_close = bool(continuation_token)
+            total_handles += response.get('number_of_handles_closed')
+            if timeout:
+                timeout = max(0, timeout - (time.time() - start_time))
+        return total_handles
```
Comment (rakshith91) [inline sdk/storage/azure-storage-file/azure/storage/file/aio/file_client_async.py]: eol...might throw a lint error 😜

File: sdk/storage/azure-storage-file/azure/storage/file/file_client.py [hunk 4]
```diff
@@ -1112,45 +1111,70 @@ def list_handles(self, **kwargs):
...
-            Optionally, a specific handle to close. The default value is '*'
-            which will attempt to close all open handles.
+            A specific handle to close.
         :type handle: str or ~azure.storage.file.Handle
         :keyword int timeout:
             The timeout parameter is expressed in seconds.
-        :returns: A long-running poller to get operation status.
-        :rtype: ~azure.core.polling.LROPoller
+        :returns:
+            The number of handles closed (this may be 0 if the specified handle was not found).
```
Comment (xiafu-msft) [inline sdk/storage/azure-storage-file/azure/storage/file/file_client.py]: handle :grinning:

Comment (annatisch) [reply sdk/storage/azure-storage-file/azure/storage/file/file_client.py]: Actually I think handles is correct in this case... although the result should only be 1 or 0...

Comment (xiafu-msft) [reply sdk/storage/azure-storage-file/azure/storage/file/file_client.py]: Sure! as long as it is intended! I just thought it will never be a plural 😀

**Pull request merged**